hydroflow_lang/src/graph/ops/fold.rs

```diff
@@ -89,8 +88,13 @@ pub const FOLD: OperatorConstraints = OperatorConstraints {
             Persistence::Tick => (
                 Default::default(),
                 quote_spanned! {op_span=>
-                    #[allow(clippy::unnecessary_fold)]
-                    let #ident = ::std::iter::once(#input.fold(#init, #func));
+                    let input = #input;
+                    let mut accum = #init;
+                    for x in input {
+                        #[allow(clippy::redundant_closure_call)]
+                        (#func)(&mut accum, x);
+                    }
+                    let #ident = ::std::iter::once(accum);
```

Comment:
put this inside `{}` and ideally use  `wc.make_ident` to prevent name collisions
```
let #ident = {
    let input = #input;
    let mut accum = #init;
    for x in input {
        #[allow(clippy::redundant_closure_call)]
        (#func)(&mut accum, x);
    }
    ::std::iter::once(accum);
};
```

Reply:
Can also reuse the existing idents, so 
```
    let #input = #input;
    let mut #ident = #init;
    for #item in #input {
        #[allow(clippy::redundant_closure_call)]
        (#func)(&mut #init, #item);
    }
    ::std::iter::once(#init);
```
though maybe confusing